robots/prowjob-experiment/handler/handler.go

```diff
@@ -0,0 +1,320 @@
+		return
+	}
+
+	gitClient, err := git.NewClient()
+	if err != nil {
+		logrus.WithError(err).Fatal("could not initialize git client")
+		return
+	}
+
+	defer gitClient.Clean()
+
+	// Disable authentication since we don't really need it
+	gitClient.SetCredentials("", func() []byte { return []byte{} })
+
+	org, name, err := gitv2.OrgRepo(event.Repo.FullName)
+	if err != nil {
+		logrus.WithError(err).Fatal("could not extract repo org and name")
+		return
+	}
+	repo, err := gitClient.Clone(org, name)
+	if err != nil {
+		logrus.WithError(err).Fatal("could not clone repo")
+		return
+	}
+
+	err = repo.CheckoutPullRequest(event.PullRequest.Number)
+	if err != nil {
+		logrus.WithError(err).Fatal("could not fetch pull request")
+		return
+	}
+
+	// TODO: handle changes to config.yaml and presets
+	modifiedJobConfigs, err := getModifiedConfigs(repo, event, jobConfigPatterns)
+	if err != nil {
+		logrus.WithError(err).Fatal("could not get the modified configs")
+		return
+	}
+
+	if len(modifiedJobConfigs) == 0 {
+		logrus.Infoln("no job configs were modified - nothing to do")
+		return
+	}
+
+	modifiedConfigs, err := loadConfigsAtRef(repo, event.PullRequest.Head.SHA, prowConfigPath, modifiedJobConfigs)
+	if err != nil {
+		logrus.WithError(err).Fatal("could not load the modified configs from pull request's head")
+		return
+	}
+
+	originalConfigs, err := loadConfigsAtRef(repo, event.PullRequest.Base.SHA, prowConfigPath, modifiedJobConfigs)
+	if err != nil {
+		logrus.WithError(err).Fatal("could not load the modified configs from pull request's base")
+		return
+	}
+
+	squashedConfigs := squashConfigPaths(originalConfigs, modifiedConfigs)
+
+	prowJobs := generateProwJobs(squashedConfigs, event)
+	createProwJobs(cl, prowJobs)
+}
+
+// Creates prow jobs
+func createProwJobs(cl *kube.Client, jobs []prowapi.ProwJob) {
+	for _, job := range jobs {
+		_, err := cl.CreateProwJob(job)
+		if err != nil {
+			logrus.WithError(err).Fatal("failed to create new ProwJob")
+		}
+	}
+}
+
+// Get the modified job configs from the repo
+func getModifiedConfigs(repo *git.Repo, event *github.PullRequestEvent, pattern string) ([]string, error) {
+	changes, err := repo.Diff(event.PullRequest.Head.SHA, event.PullRequest.Base.SHA)
+	if err != nil {
+		return nil, err
+	}
+	return filterByPattern(changes, pattern)
+}
+
+// Filter the input array/slice by the given pattern.
+// The pattern has to be a shell file name pattern.
+// https://golang.org/pkg/path/filepath/#Match
+func filterByPattern(input []string, pattern string) ([]string, error) {
+	var out []string
+	for _, s := range input {
+		match, err := filepath.Match(pattern, s)
+		if err != nil {
+			// The only possible error is ErrorBadPattern
+			return nil, err
+		}
+		if match {
+			out = append(out, s)
+		}
+	}
+	return out, nil
+}
+
+// Determine if we should perform on the given event.
+func shouldPerformOnEventAction(event *github.PullRequestEvent) bool {
```

Comment:
This method name does convey that it checks whether an action should be performed. However, to check in what case an action will be performed you need to read the method body. I'd recommend renaming the method to better convey that the following code should only run if a PR was opened, edited or synced. 

In short I mean, the method name talks about how it does things, but not what it does.

This is not an easy topic for me to explain, hopefully this makes sense to you.